### PR TITLE
Add exploit for CVE-2016-6267 - Trend Micro Smart Protection Server authenticated RCE.

### PR DESCRIPTION
**Pull request by qkaiser**
Name: Trend Micro Smart Protection Server Exec Remote Code Injection
Module: exploit/linux/http/trendmicro_sps_exec
Platform: Linux
Privileged: No
License: Metasploit Framework License (BSD)
Disclosed: 2016-08-08

This module exploits a vulnerability found in TrendMicro Smart Protection Server where untrusted inputs are fed to ServWebExec system command, leading to command injection. Please note that authentication is required to exploit this vulnerability.

Advisory: http://qkaiser.github.io/pentesting/trendmicro/2016/08/08/trendmicro-sps/

### CODE REVIEW

**Comment by wwebb-r7** [comment]
This probably would have been landed over a month ago, but obtaining the software for testing has so far been a pain.

**Comment by stevenseeley** [comment]
Actually, that was pretty easy to validate.

**Comment by stevenseeley** [comment]
Actually, that was pretty easy to validate.

**Comment by busterb** [comment]
Cool, did it work for you @stevenseeley ?

**Comment by stevenseeley** [comment]
@busterb like a charm, but module design needs to be a little improved.

File: modules/exploits/linux/http/trendmicro_safesync_exec.rb [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+        if (version <= 2.1 and build_major == 3 and build_minor == 2 and build_version == 0 and build_tag < 1426)
+            return Exploit::CheckCode::Vulnerable
+        else
+          return Exploit::CheckCode::Safe
+        end
+      end
+    end
+    Exploit::CheckCode::Safe
+  end
```
Comment (bwatters-r7) [inline modules/exploits/linux/http/trendmicro_safesync_exec.rb]: I'm confused by the double Exploit::Checkcode::Safe.  Is there a reason for it?  Should the second one be unknown, maybe?  I feel like it should be unknown.

File: modules/exploits/linux/http/trendmicro_sps_exec.rb [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+        if (version == 3.0 and build < 1330) or
+          (version == 2.6 and build < 2106) or
+          (version == 2.5 and build < 2200)
+            return Exploit::CheckCode::Vulnerable
+        else
+          return Exploit::CheckCode::Safe
+        end
+      end
+    end
+    Exploit::CheckCode::Safe
```
Comment (bwatters-r7) [inline modules/exploits/linux/http/trendmicro_sps_exec.rb]: Same double-safe, even if the response is not as expected.

Comment (QKaiser) [inline modules/exploits/linux/http/trendmicro_sps_exec.rb]: You're right. Second one should be unknown. Just fixed it :)

File: modules/exploits/linux/http/trendmicro_sps_exec.rb [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+        }
+      })
+      if res and res.code == 200
+        version = res.body.to_s.scan(/MSG_ABOUT_VERSION <\/td>[^<]*<td[^>]*>([^<]*)</).last.first.to_f
+        build = res.body.to_s.scan(/MSG_ABOUT_BUILD <\/td>[^<]*<td[^>]*><span[^>]*>([^<]*)</).last.first.to_i(10)
+        print_status("TrendMicro Smart Protection Server detected.")
+        print_status("Version: #{version}")
+        print_status("Build: #{build}")
+        if (version == 3.0 and build < 1330) or
+          (version == 2.6 and build < 2106) or
```
Comment (h00die) [inline modules/exploits/linux/http/trendmicro_sps_exec.rb]: would these version checks be more simple with a Gem::version?

**Comment by wchen-r7** [comment]
I thought I had the vulnerable software to test, but turns out I don't. It looks like I am not able to verify this PR for now. I would like to unassign myself to avoid hijacking the PR. If anybody has an idea where to get the vulnerable software (trial?), please let me know. Thanks!

**Comment by jmartin-r7** [comment]
# Release Notes

This module exploits a vulnerability found in TrendMicro Smart Protection Server where untrusted inputs are fed to ServWebExec system command, leading to command injection. Please note that authentication is required to exploit this vulnerability. Unpatched versions 2.5, 2.6, and 3.0 are vulnerable.

**Pull request merged**